src/Sema.zig

```diff
@@ -22516,14 +22540,22 @@ const CoersionError = CompileError || error{
     NotCoercible,
 };
 
+const CoerceOpts = struct {
+    /// Should coerceExtra emit error messages.
+    report_err: bool = true,
+    /// Ignored if `report_err == false`.
+    is_ret: bool = false,
+    /// Should coercion to comptime_int ermit an error message.
+    no_cast_to_comptime_int: bool = false,
```

Comment:
I don't think this behavior should be different with `const y = @as(T, x)` vs `const y: T = x;`. We want to be able to say that these two statements are identical. I'd like to push back on adding `ty_builtin` as a ResultLoc, and `builtin_as` as a ZIR instruction.

Reply:
There needs to be some way to disable that error for casting the shift operand to the log2 type in order to get a good error message. Would it be better if there was instead a `as_shift_operand` instruction?